lib/driverProviders/driverProvider.ts

```diff
@@ -89,42 +88,38 @@ export abstract class DriverProvider {
 
 
   /**
-   * Quits an array of drivers and returns a q promise instead of a webdriver one
+   * Quits an array of drivers and returns a native promise instead of a webdriver one
    *
    * @param drivers {webdriver.WebDriver[]} The webdriver instances
    */
-  static quitDrivers(provider: DriverProvider, drivers: WebDriver[]): q.Promise<void> {
-    let deferred = q.defer<void>();
-    wdpromise
-        .all(drivers.map((driver: WebDriver) => {
-          return provider.quitDriver(driver);
-        }))
-        .then(
-            () => {
-              deferred.resolve();
-            },
-            () => {
-              deferred.resolve();
-            });
-    return deferred.promise;
+  static quitDrivers(provider: DriverProvider, drivers: WebDriver[]): Promise<void> {
+    let deferredResolve: (x?: any) => void;
+    let deferredReject: (x?: any) => void;
+    let deferred = new Promise<void>((resolve, reject) => {
+      deferredResolve = resolve;
+      deferredReject = reject;
+    });
+    wdpromise.all(drivers.map((driver: WebDriver) => provider.quitDriver(driver)))
+        .then(() => deferredResolve(), () => deferredReject());
```

Comment:
The WebDriver promises have different characteristics, I believe, and are tied into ControlFlow, which I have yet to deeply familiarize myself with.

As such, propagating them through the system might magically improve everything, or might be a complete non-starter, when considering the concurrency stories.
The additional anonymous functions wrapping the resolve and reject were to guarantee that nothing was passed on, in the instance that the method signature of `quitDriver` never changes, or to equally maintain the behaviour of ignoring the error thrown.

My intent is to look deeper into these cases, but due to the nature of how these things are set up, changing them may lead to rather large refactorings (which I'm not opposed to; I just didn't want to juggle it, while also worrying about changing libraries).